File: frontend/src/types/analysis.ts
export interface AnalysisRequest {
  user_id: number;

  image_name: string;

  crop_type: string;

  quality: string;

  stage: string;
  stage_confidence: number;

  stress_class: string;
  stress_confidence: number;

  severity: number;
  severity_label: string;

  latitude: number;
  longitude: number;
}

export interface AnalysisResponse extends AnalysisRequest {
  success?: boolean;
  message?: string;
  instruction?: string;
  ai_explanation?: string;
  id: number;
  created_at: string;
}

File: app-backend/app/schemas/analysis.py
from datetime import datetime

from pydantic import BaseModel, Field


class AnalysisCreate(BaseModel):

    ai_explanation: str | None = Field(default=None, max_length=6000)

    user_id: int

    image_name: str

    crop_type: str

    quality: str

    stage: str
    stage_confidence: float

    stress_class: str
    stress_confidence: float

    severity: float
    severity_label: str

    latitude: float
    longitude: float


class AnalysisResponse(AnalysisCreate):

    id: int

    created_at: datetime

    class Config:
        from_attributes = True